Missing comma in try_wait() in __cuda/barrier.h

https://github.com/NVIDIA/cccl/blob/2bd685f0d524be3565d65bd93d2fc02ff9ce4d42/libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h#L346

```
                    asm volatile ("{\n\t"
                            ".reg .pred p;\n\t"
                            "mbarrier.try_wait.shared.b64 p, [%1], %2, %3;\n\t"
                            "selp.b32 %0, 1, 0, p;\n\t"
                            "}"
                            : "=r"(__ready)
                            : "r"(static_cast<_CUDA_VSTD::uint32_t>(__cvta_generic_to_shared(&__barrier))),
                            "l"(__token)
                            "r"(__wait_nsec)
                            : "memory");
```

There must be a comma between `__token` and `__wait_nsec` arguments.

Found this trying to compile the file with clang:

```
In file included from third_party/gpus/cccl/libcudacxx/test/libcudacxx/cuda/barrier/expect_tx_device.runfail.cpp:18:
In file included from third_party/gpus/cccl/libcudacxx/include/cuda/barrier:14:
In file included from third_party/gpus/cccl/libcudacxx/include/cuda/std/barrier:22:
In file included from third_party/gpus/cccl/libcudacxx/include/cuda/std/detail/libcxx/include/barrier:484:
third_party/gpus/cccl/libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h:343:29: error: expected ')'
  343 |                             "r"(__wait_nsec)
      |                             ^
1 error generated when compiling for sm_90.
```

If the `expect_tx_device.runfail.cpp` test is not failing in CI, there may be a gap in test coverage there.

--- a/libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h
+++ b/libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h
@@ -343,7 +343,7 @@ private:
                             "}"
                             : "=r"(__ready)
                             : "r"(static_cast<_CUDA_VSTD::uint32_t>(__cvta_generic_to_shared(&__barrier))),
-                            "l"(__token)
+                            "l"(__token),
                             "r"(__wait_nsec)
                             : "memory");
                     __elapsed = _CUDA_VSTD::chrono::high_resolution_clock::now() - __start;

